app/src/main/java/net/bible/android/view/activity/readingplan/ReadingPlanSelectorList.kt

```diff
@@ -73,23 +70,27 @@ class ReadingPlanSelectorList : ListActivityBase() {
         Log.d(TAG, "Finished displaying Reading Plan list")
     }
 
-    /** if a plan is selected then ask confirmation, save plan, and go straight to first day
-     */
     override fun onListItemClick(l: ListView, v: View, position: Int, id: Long) {
         try {
-            readingPlanControl.startReadingPlan(mReadingPlanList[position])
+            dbAdapter.switchToReadingPlan(readingPlanList[position])
+
+            val readingPlanOneDay: ReadingPlanOneDayDB? = try {
+                ReadingPlanOneDayDB(readingPlanList[position])
+            } catch (e: Exception) { null }
+
+            ABEventBus.getDefault().post(ReadingPlanDayChangeEvent(readingPlanOneDay, readingPlanList[position]))
 
-            val intent = Intent(this@ReadingPlanSelectorList, DailyReading::class.java)
-            startActivity(intent)
+            setResult(Activity.RESULT_OK)
             finish()
+
         } catch (e: Exception) {
             Log.e(TAG, "Plan selection error", e)
             Dialogs.getInstance().showErrorMsg(R.string.error_occurred, e)
         }
```

Comment:
(Again) I would like us to re-consider if this is such a place in our code, where we are expecting errors to take place easily (because of user mistake for example -- for example error in reading plan document)